crates/dids/methods/ethr/src/lib.rs

```diff
@@ -487,7 +487,7 @@ mod tests {
         let did = DIDEthr::generate(&key).unwrap();
         eprintln!("did: {}", did);
 
-        let cred = JsonCredential::new(
+        let cred: JsonCredential = JsonCredential::new(
```

Comment:
This is confusing and might not generate nice errors -- is there no way to avoid that?

Reply:
I don't like that either. Previously I had a `SpecializedJsonCredential` with type parameters and `JsonCredential` with sensible defaults to avoid having to use type annotations.

Reply:
Maybe I can have an in-between solution, where I reserve the `new` constructor for default type parameters, and add a `new_specialized` constructor with custom type parameters.